rest/attachmentcompactiontest/attachment_compaction_api_test.go

```diff
@@ -156,6 +157,113 @@ func TestAttachmentCompactionAPI(t *testing.T) {
+	}
+	var garbageVBUUID gocbcore.VbUUID = 1234
+
+	rt := rest.NewRestTesterDefaultCollection(t, nil)
+	defer rt.Close()
+	dataStore := rt.GetSingleDataStore()
+
+	// Create some 'unmarked' attachments
+	makeUnmarkedDoc := func(docid string) {
+		err := dataStore.SetRaw(docid, 0, nil, []byte("{}"))
+		require.NoError(t, err)
+	}
+
+	for i := 0; i < 1000; i++ {
+		docID := fmt.Sprintf("%s%s%d", base.AttPrefix, "unmarked", i)
+		makeUnmarkedDoc(docID)
+	}
+
+	// kick off compaction and wait for "mark" phase to begin
+	resp := rt.SendAdminRequest("POST", "/{{.db}}/_compact?type=attachment", "")
+	rest.RequireStatus(t, resp, http.StatusOK)
+
+	var response db.AttachmentManagerResponse
+	err := rt.WaitForCondition(func() bool {
+		time.Sleep(1 * time.Second)
+
+		resp := rt.SendAdminRequest("GET", "/{{.db}}/_compact?type=attachment", "")
+		rest.RequireStatus(t, resp, http.StatusOK)
+
+		err := base.JSONUnmarshal(resp.BodyBytes(), &response)
+		require.NoError(t, err)
+
+		return response.Phase == "mark"
+	})
+	require.NoError(t, err)
+
+	// stop the mark phase before completion
+	resp = rt.SendAdminRequest("POST", "/{{.db}}/_compact?type=attachment&action=stop", "")
+	rest.RequireStatus(t, resp, http.StatusOK)
+	err = rt.WaitForCondition(func() bool {
+		time.Sleep(1 * time.Second)
+
+		resp := rt.SendAdminRequest("GET", "/{{.db}}/_compact?type=attachment", "")
+		rest.RequireStatus(t, resp, http.StatusOK)
+
+		err := base.JSONUnmarshal(resp.BodyBytes(), &response)
+		require.NoError(t, err)
+
+		return response.State == db.BackgroundProcessStateStopped
+	})
+	require.NoError(t, err)
+
+	// alter persisted dcp metadata from the first run to force a rollback
+	name := db.GenerateCompactionDCPStreamName(response.CompactID, "mark")
+	checkpointPrefix := fmt.Sprintf("%s:%v", "_sync:dcp_ck:", name)
+
+	meta := base.NewDCPMetadataCS(dataStore, 1024, 8, checkpointPrefix)
+	vbMeta := meta.GetMeta(0)
+	vbMeta.VbUUID = garbageVBUUID
+	meta.SetMeta(0, vbMeta)
+	meta.Persist(0, []uint16{0})
+
+	// kick off a new run attempting to start it again (should force into rollback handling)
+	resp = rt.SendAdminRequest("POST", "/{{.db}}/_compact?type=attachment&action=start", "")
+	rest.RequireStatus(t, resp, http.StatusOK)
+	err = rt.WaitForCondition(func() bool {
+		time.Sleep(1 * time.Second)
+
+		resp := rt.SendAdminRequest("GET", "/{{.db}}/_compact?type=attachment", "")
+		rest.RequireStatus(t, resp, http.StatusOK)
+
+		err := base.JSONUnmarshal(resp.BodyBytes(), &response)
+		require.NoError(t, err)
+
+		return response.State == db.BackgroundProcessStateRunning
+	})
+	require.NoError(t, err)
+
+	err = rt.WaitForCondition(func() bool {
+		time.Sleep(1 * time.Second)
+
+		resp := rt.SendAdminRequest("GET", "/{{.db}}/_compact?type=attachment", "")
+		rest.RequireStatus(t, resp, http.StatusOK)
+
+		err = base.JSONUnmarshal(resp.BodyBytes(), &response)
+		require.NoError(t, err)
+
+		return response.State == db.BackgroundProcessStateCompleted
+	})
```

Comment:
```suggestion
	rt.WaitForAttachmentCompactionStatus(db.BackgroundProcessStopped)
```

I'm reasonably certain there's a test race condition if this runs too fast we'll miss the `Running` status.